src/Server.cpp

```diff
@@ -1,182 +1,188 @@
 
-/* ====== HELPER FUNCTIONS ====== */
-
-/**
- * @brief Set the socket to nonblocking mode
- * 
- * Use fcntl to manipulate fd/sockets. Retrieve the currently set flags with
- * F_GETFL. Set the flag with with F_SETFL. Use bitwise OR to set the 
- * O_NONBLOCK bit in "flags" while preserving all other bits.
- * @param sock	Socket to be manipulated
- */
-static void setNonblocking(int sock)
-{
-	int	flags = fcntl(sock, F_GETFL, 0);
-	if (flags == -1) {
-		throw std::runtime_error("fcntl(F_GETFL)");
-	}
-	if (fcntl(sock, F_SETFL, flags | O_NONBLOCK) == -1) {
-		throw std::runtime_error("fcntl(F_SETFL)");
-	}
-}
 /* ====== CONSTRUCTOR/DESTRUCTOR ====== */
 
 /**
  * @brief Constructor for the Server class.
- * 
+ *
  * This constructor initializes a server instance by creating a server socket,
- * setting it to non-blocking mode, binding it to a specific address and port,
- * listening on the server socket, creating an epoll instance for event handling,
- * and adding the server socket to the epoll instance.
- * 
+ * using TCP protocol SOCK_STREAM, and setting it to non-blocking mode.
+ * It also creates an epoll instance for event handling.
+ * If any of these operations fail, the constructor throws a std::runtime_error.
+ * Then it bind() the server socket to a specific address and port, listen() on the server socket
+ * Then the server socket is added to the epoll instance.
+ *
  * @throws std::runtime_error if any of the socket creation, binding, listening,
  *         epoll creation, or epoll control operations fail.
- * 
+ *
  */
 Server::Server()
+	: m_serverSock(socket(AF_INET, SOCK_STREAM | SOCK_NONBLOCK, 0))
+	, m_epfd(epoll_create(1))
 {
-    // Create server socket using TCP protocol SOCK_STREAM
-	m_serverSock = socket(AF_INET, SOCK_STREAM, 0);
 	if (m_serverSock < 0) {
 		throw std::runtime_error("socket");
 	}
 
-	// Set server socket to non-blocking
-	setNonblocking(m_serverSock);
+	if (m_epfd < 0) {
+		close(m_serverSock);
+		throw std::runtime_error("epoll_create");
+	}
 
 	// Bind server socket
-    struct sockaddr_in	server_addr = {};
-	server_addr.sin_family = AF_INET; // Communicates with IPv4
-	server_addr.sin_addr.s_addr = INADDR_ANY; // Accept any arriving ip addresses
-	server_addr.sin_port = htons(PORT); // Listen on specified port
-	if (bind(m_serverSock, (struct sockaddr*)&server_addr, sizeof(server_addr))
-		< 0) {
-  		close(m_serverSock);
+	struct sockaddr_in serverAddr = {};
+	serverAddr.sin_family = AF_INET; // Communicates with IPv4
+	serverAddr.sin_addr.s_addr = INADDR_ANY; // Accept any arriving ip addresses
+	serverAddr.sin_port = htons(PORT); // Listen on specified port
+	if (bind(m_serverSock, reinterpret_cast<struct sockaddr*>(&serverAddr), sizeof(serverAddr)) < 0) { // NOLINT
+		close(m_serverSock);
 		throw std::runtime_error("bind");
 	}
 
 	// Listen on server socket
-	if (listen(m_serverSock, 10) < 0) {
-        close(m_serverSock);
+	if (listen(m_serverSock, CONNECTION_QUEUE) < 0) {
```

Comment:
I'm not convinced. 
I think we should change it to a const int. But I'm not sure where yet, since the server class will be refactored out, I suspect (since we can have more than one server but only one epoll)
See also PR #20 and core guidelines rule ES.31: [Don’t use macros for constants or “functions”](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es31-dont-use-macros-for-constants-or-functions)

Reply:
Interesting that they are so opposed to macros...